### PR TITLE
MINOR: convert GssapiAuthenticationTest to KRaft

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala [hunk 1]
```diff
@@ -92,15 +92,16 @@ class GssapiAuthenticationTest extends IntegrationTestHarness with SaslSetup {
    * Tests that Kerberos replay error `Request is a replay (34)` is not handled as an authentication exception
    * since replay detection used to detect DoS attacks may occasionally reject valid concurrent requests.
    */
-  @Test
-  def testRequestIsAReplay(): Unit = {
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
+  def testRequestIsAReplay(quorum: String): Unit = {
```
Comment (mjsax) [inline core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala:97]: Why is `quorum` an input parameter to this test? Seems the value would always be `"kraft"`? -- I also also don't see where `quorum` is actually used?

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala [hunk 2]
```diff
@@ -109,7 +110,8 @@ class GssapiAuthenticationTest extends IntegrationTestHarness with SaslSetup {
    * are able to connect after the second re-login. Verifies that logout is performed only once
    * since duplicate logouts without successful login results in NPE from Java 9 onwards.
    */
-  @Test
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
```
Comment (mjsax) [inline core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala:114]: There is no input parameter to `testLoginFailure()` -- why change this to being a parametrized test? (Same below.)

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala (hunk 1 above)
Comment (cmccabe) [reply core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala:97]: While the parameter isn't used directly, it affects the test name which some code in `TestUtils.scala` uses to determine if the test is kraft or not. This is standard for our kraft tests

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala [hunk 3]
```diff
@@ -109,7 +110,8 @@ class GssapiAuthenticationTest extends IntegrationTestHarness with SaslSetup {
...
   def testLoginFailure(): Unit = {
```
Comment (cmccabe) [inline core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala]:
```suggestion
  def testLoginFailure(quorum: String): Unit = {
```

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala [hunk 4]
```diff
@@ -163,7 +166,8 @@ class GssapiAuthenticationTest extends IntegrationTestHarness with SaslSetup {
    * Tests that Kerberos error `Server not found in Kerberos database (7)` is handled
    * as a fatal authentication failure.
    */
-  @Test
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
   def testServerNotFoundInKerberosDatabase(): Unit = {
```
Comment (cmccabe) [inline core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala]:
```suggestion
  def testServerNotFoundInKerberosDatabase(quorum: String): Unit = {
```

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala (hunk 2 above)
Comment (cmccabe) [reply core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala:114]: This was supposed to take a parameter. I'll fix it (and the one below). See the explanation above.

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala [hunk 5]
```diff
@@ -132,7 +134,8 @@ class GssapiAuthenticationTest extends IntegrationTestHarness with SaslSetup {
    * is performed when credentials are unavailable between logout and login, we handle it as a
    * transient error and not an authentication failure so that clients may retry.
    */
-  @Test
+  @ParameterizedTest
+  @ValueSource(strings = Array("kraft"))
```
Comment (jolshan) [inline core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala:138]: Should this also have the quorum param?

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala [hunk 6]
```diff
@@ -132,7 +134,8 @@ class GssapiAuthenticationTest extends IntegrationTestHarness with SaslSetup {
...
   def testReLogin(): Unit = {
```
Comment (cmccabe) [inline core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala]:
```suggestion
  def testReLogin(quorum: String): Unit = {
```

File: core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala (hunk 5 above)
Comment (cmccabe) [reply core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala:138]: fixed :)

**Review by jolshan — APPROVED** [review]
lgtm as long as the tests pass. Maybe at some point we can refactor to not need the quorum param and just always use kraft. But this is good for this change 👍

**Review by mjsax — APPROVED** [review]

**Pull request merged**